extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -465,16 +504,42 @@ export class AmpLightboxViewer extends AMP.BaseElement {
     this.updateInViewport(dev().assertElement(this.container_), true);
     this.scheduleLayout(dev().assertElement(this.container_));
 
-    this.currentElementId_ = element.lightboxItemId;
-    // Hack to access private property. Better than not getting
-    // type checking to work.
-    /**@type {?}*/ (this.carousel_).implementation_.showSlideWhenReady(
-        this.currentElementId_);
-
     this.win.document.documentElement.addEventListener(
         'keydown', this.boundHandleKeyboardEvents_);
 
-    return Promise.resolve();
+    if (this.carousel_) {
```

Comment:
not new code, but `dev().assert(this.carousel_)` is the right way to handle "should never happen or is a bug" cases.